x/evm/keeper/grpc_query.go

```diff
@@ -361,9 +361,16 @@ func (k Keeper) TraceTx(c context.Context, req *types.QueryTraceTxRequest) (*typ
 	if req.TraceConfig != nil && req.TraceConfig.Limit < 0 {
 		return nil, status.Errorf(codes.InvalidArgument, "output limit cannot be negative, got %d", req.TraceConfig.Limit)
 	}
+	//
+	// minus one to get the context of block beginning
+	contextHeight := req.BlockNumber - 1
```

Comment:
I think no need to minus one here, because the store is already at the beginning of the block, and `ctx.WithBlockHeight` is to modify the tx context, so `block.number` in smart contract will see the correct block number.

Reply:
The weird thing, is that when I dont do it the transactions fail as if the sender user had been deducted the balance already. It starts with the balance after txs and not before them